server/server.go

```diff
@@ -307,7 +305,7 @@ func (s *Server) GetConfig(instID flux.InstanceID, includeSecrets bool) (flux.In
 
 	config := fullConfig.Settings
 	if !includeSecrets {
-		removeSecrets(&config)
+		config = config.HideSecrets()
```

Comment:
I hear ya -- safety by default. Maybe some tricky thing whereby the usual config value marshals with hidden secrets, but you can get another type that marshals *with* secrets?

(cross fingers there ends up being no good reason to ever transmit secrets, in which case this goes away)